### PR TITLE
Pull Request for Issue1737: SGM Continuous Energy Scan Configuration

### PR DESCRIPTION
**Pull request by dretrex**
This takes the code out of the continuous scan action controller and puts it into a specific XAS scan controller/configuration/view triplet.  I modified the code inside AMContinuousScanController to provide some virtual functions that subclasses can implement for the specific needs of the continuous axis.

I also added a new set of exposed detectors in AMBeamline called `exposedScientificDetectors()`.

### CODE REVIEW

File: source/acquaman/AMContinuousScanActionController.h [hunk 1]
```diff
@@ -4,17 +4,16 @@
 #include "acquaman/AMScanActionController.h"
 #include "acquaman/AMContinuousScanConfiguration.h"
 #include "dataman/AMUser.h"
+#include "acquaman/AMAgnosticDataAPI.h"
+#include "beamline/SGM/energy/SGMGratingSupport.h"
```
Comment (davidChevrier) [inline source/acquaman/AMContinuousScanActionController.h]: Is this needed here anymore?

File: source/application/SGM/SGMAppController.cpp [hunk 2]
```diff
@@ -224,34 +226,44 @@ void SGMAppController::setupUserInterface()
...
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("FilteredPD4")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("HexapodRed")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("HexapodBlack")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("EncoderUp")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("EncoderDown")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("AmptekSDD1")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("AmptekSDD2")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("AmptekSDD3")->toInfo());
+	xasScanConfiguration_->addDetector(SGMBeamline::sgm()->exposedDetectorByName("AmptekSDD4")->toInfo());
+	AMDetectorSet *xasScanConfigurationDetectors = new AMDetectorSet(this);
```
Comment (davidChevrier) [inline source/application/SGM/SGMAppController.cpp]: Was this detector set a stepping stone before going to the exposedScientificDetectors() paradigm? Is it needed now?

File: source/acquaman/AMContinuousScanActionController.h (hunk 1 above)
Comment (dretrex) [inline source/acquaman/AMContinuousScanActionController.h]: Not sure, I just wanted to make sure all the includes were at the top.

**Comment by davidChevrier** [comment]
Comments made. Let me know if or how you want to fix them up.
Time Estimate: 10 minutes

File: source/application/SGM/SGMAppController.cpp (hunk 2 above)
Comment (dretrex) [inline source/application/SGM/SGMAppController.cpp]: Yup.  I'll remove that.

**Comment by dretrex** [comment]
Time Estimate: 5 minutes

**Pull request merged**